File: 189-rotate-array/189-rotate-array.py
class Solution:
    def reverse(self, arr, a, b):
        while a <= b:
            temp = arr[a]
            arr[a] = arr[b]
            arr[b] = temp
            a += 1 
            b -= 1
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        if k > len(nums):
            k = k % len(nums)
        self.reverse(nums, n-k, n-1)
        self.reverse(nums, 0, n-k-1)
        self.reverse(nums, 0, n-1)
        print(nums)

File: 189-rotate-array/189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int count = k % nums.size();
        if(count==0)
            return;
        reverse(nums.begin(), nums.end() - count);
        reverse(nums.end() - count, nums.end());
        reverse(nums.begin(), nums.end());
        return;
    }
};